schedule/schedule-appengine/src/main/client/jasify/admin/activity/admin-activites.js

```diff
@@ -38,6 +43,22 @@
             $location.path('/admin/activities/' + organization.id);
         }
 
+        function queryActivities() {
+            if (vm.organization) {
+                var dfd = $q.defer();
+
+                Activity.query({
+                    organizationId: vm.organization.id,
+                    fromDate: vm.activitiesQueryFromDate
+                }).then(function (result) {
+                    dfd.resolve(angular.extend({items: []}, result));
+                    vm.activities = result.items;
+                });
+            } else {
+                return {items: []};
+            }
+        }
+
```

Comment:
Maybe a note on the whole method...  What you want to do is
```javascript
function queryActivities() {
  vm.activites = []; //Since you will query, you want to clear the screen
   //This is a promise since query() is async.  when the response from the server comes back it will run "ok" or "nok" 
 Activities.query().then(ok, nok);
 function ok(r) {   vm.activites = r.items; }
 function nok(r) { jasDialogs.error(....); }
```